### ISSUE TITLE
Question : can this api be used to produce snapchat like masks and filters ? 

### ISSUE THREAD

**Issue by usman-ghauri** [googlesamples/android-vision#101]
So i recently tested out this api and run sample code. is there any way i can use this api to integrate those masks features in face tracking sample. 

What i already did is that i draw a resource drawable as an overlay on the rectangle bounds tracked by face tracker. and when i capture image from it it only gives me camera picture withoutu overlay. i need to capture it with overlay / mask

**Comment by guyzk** [comment]
The only way I have found how to record a video with an overlay (without external libs such as OpenCV or FFmpeg) is to use a GLSurface + camera face tracking. Recording from the surface using MediaCodec and MediaMuxer and drawing to the GL surface any overlay you want.
Not so simple (You need some OpenGL understanding), but this seems like a start...

**Comment by PopTudor** [comment]
yes it can be used but the performance is not so great and the models lag behind. I used Rajawali to move/scale/translate the 3D objects and I'm not sure if it lags because I'm doing too many of those operations or because of the library. However snapchat/MSQRD use opencv and other specialised algorithms. Take a look at https://github.com/TadasBaltrusaitis/OpenFace but it's not yet implemented/ported to java/android and there are no metrics about memory usage on mobile (one guy made it work on iOS though). There is also another alternative that is a bit more straightforward: https://github.com/tzutalin/dlib-android . I'm trying to port my app from mobile vision to this library to see how it performs. You can check it out the app here https://play.google.com/store/apps/details?id=com.marked.pixsee . Make a test account and grand camera permission when asked or else it crashes
